ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java

```diff
@@ -116,17 +114,17 @@ private void deleteTopics(final List<String> topicsToBeDeleted) {
   }
 
   private void cleanUpSinkAvroSchemas(final List<String> topicsToBeDeleted) {
-    final MetaStore metaStore = ksqlEngine.getMetaStore();
-    final Stream<StructuredDataSource> avroSourcesToCleanUp = topicsToBeDeleted.stream()
-        .map(metaStore::getSourceForTopic)
-        .filter(Optional::isPresent)
-        .map(Optional::get)
-        .filter(dataSource -> dataSource.isSerdeFormat(DataSourceSerDe.AVRO));
-    final Stream<String> subjectsToDelete = avroSourcesToCleanUp
-        .map(source -> source.getName() + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
+    final Stream<String> subjectsToDelete = topicsToBeDeleted.stream()
+        .filter(this::hasAvroSource)
+        .map(topicName -> topicName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
     filterNonExistingSubjects(subjectsToDelete).forEach(this::deleteSubject);
   }
 
+  private boolean hasAvroSource(final String topicName) {
+    return ksqlEngine.getMetaStore().getSourcesForKafkaTopic(topicName).stream()
+        .anyMatch(dataSource -> dataSource.isSerdeFormat(DataSourceSerDe.AVRO));
```

Comment:
This is a little overkill since if there are multiple streams defined on top of the same topic, they should all have the same format. Shouldn't hurt either though.